### PR TITLE
🏗  Allow story examples to be excluded from amp validation

### PR DESCRIPTION
**Pull request by mszylkowski**
We want to modify the `htmlFixtureGlobs` so when we create new example stories (which are usually invalid), we don't need to ask for OWNERS approvals outside of our regular approvers.

Followup to: #34961

eg: https://github.com/ampproject/amphtml/pull/35002

### CODE REVIEW

**Review by rcebulko — COMMENTED** [review]
Would it make more sense for the `htmlFixtureGlobs` to just have an exclusion for stories a la `!examples/{,visual-tests/}amp-story/**/*.html`? Giving everyone OWNERS to all of the test configs seems like overkill here

**Comment by rsimha** [comment]
> Would it make more sense for the `htmlFixtureGlobs` to just have an exclusion for stories a la `!examples/{,visual-tests/}amp-story/**/*.html`? Giving everyone OWNERS to all of the test configs seems like overkill here

Good point. @mszylkowski can you comment on which of the fixtures in `examples/amp-story/` are supposed to be valid AMP? If it's none of them, Ryan's idea will work great. If it's some of them, could they be moved to a separate directory so the invalid ones can be ignored?

**Comment by mszylkowski** [comment]
Sounds good give that there's other configs in the file. I'm not sure how often the other teams create new examples, but adding our folders to the exclusion list works for us.

File: build-system/test-configs/config.js [hunk 1]
```diff
@@ -263,48 +263,7 @@ const htmlFixtureGlobs = [
...
-  '!examples/amp-story/show-tooltip.html',
-  '!examples/amp-story/supports-landscape-ads.html',
-  '!examples/amp-story/text-background-color.html',
-  '!examples/amp-story/video-one-page.html',
-  '!examples/amp-story/video-one-page2.html',
-  '!examples/amp-story/videos-cdn.html',
-  '!examples/amp-story/videos-google-cache.html',
-  '!examples/amp-story/videos.html',
-  '!examples/amp-story/visual-effects.html',
+  '!examples/{,visual-tests/}amp-story/**/*.html',
```
Comment (rsimha) [inline build-system/test-configs/config.js:266]: Two questions:
1. If these aren't meant to be valid AMP, they should be moved from this "to fix" list to the "always invalid" list above.
2. Why are we including `visual-tests`? IIUC, all visual tests are run against valid AMP pages. Unless I'm wrong?

Comment (rcebulko) [reply build-system/test-configs/config.js:266]: I based the glob suggestion on the fact that there many existing visual-diff examples of amp-story. Not sure if that's intended or not.

Comment (rsimha) [reply build-system/test-configs/config.js:266]: On the same lines, there are E2E tests for stories in the ignore list as well. Should they be ignore too?

Comment (mszylkowski) [reply build-system/test-configs/config.js:266]: I don't see any visual tests in this list, so we can not include them in this TODO list. Then adding the `examples/amp-story` to the invalid docs sounds good.

Comment (mszylkowski) [reply build-system/test-configs/config.js]: In general I see a couple of things that don't make sense to have in this list, so I can check those things and then re-request the review. For instance, `amp-story-player` documents are not meant to be AMP documents in many cases, as the player works on non-AMP docs

Comment (rcebulko) [reply build-system/test-configs/config.js]: Perhaps even better, `foobar.invalid.html` naming convention?

Comment (rsimha) [reply build-system/test-configs/config.js]: I'm pretty sure `globby` (which uses `fast-glob`) does not support positive or negative globs for non-matches (e.g. all files that _don't_ end with `.amp.html`). Using a `invalid.html` convention SGTM. 👍

Comment (rcebulko) [reply build-system/test-configs/config.js]: Globby does support excluding matches. I'm not sure exactly what you're trying to do, but ex. 
`globby.sync(['examples/**/*.html', '!examples/**/*.amp.html'])` works as you'd expect

Comment (rsimha) [reply build-system/test-configs/config.js]: Yes, but I don't think it supports excluding non-matches. E.g. _"ignore every file under `foo/` directory that does not match `bar` pattern"_.

Comment (mszylkowski) [reply build-system/test-configs/config.js]: I made it work already by excluding `<path>/!(*.amp.html)`, so essentially exclude the amp.html from the exclusion. Will update the PR soon.

**Comment by mszylkowski** [comment]
@rcebulko @rsimha Added the couple of folders that we mostly use into a glob pattern that only checks the `.amp.html` files. PTAL.

Bringing the pattern of only checking for the `.amp.html` instead of `.html` could be a good idea across the codebase, but for us it's great to be able to ensure validation only on the files that we want.

**Review by rsimha — APPROVED** [review]
@mszylkowski Glad this was resolved, and thanks to you and @rcebulko for patiently dealing with my questions 😃

**Pull request merged**